docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,112 @@
+policy, and adminstrators may desire to use it instead of the one provided with aggregated logging.  It is assumed in this configuration, the APIMan management interface will
+additionally rely on Kibana deployed off-cluster.  The multi-tenant features provided by aggregated logging for ElasticSearch and Kibana may not be available when configured to utilize these off-cluster
+dependencies.      
+
+## Use Cases
+* **UC01** As an API provider, I want to navigate from the OpenShift web console to the policy management interface, so I can define my service policies.
+* <s> **UC02** As an API provider, I want to associate an APIMan organization with an OpenShift project, so I can control policy between organizations.</s>
+* **UC03** As an API provider, I want to expose my service, so that it is consumable according to my management policy.  
+* **UC04** As a cluster administrator, I want a gateway to route API traffic only, so that it manages traffic based on a service's policy.
+* **UC05** As a cluster administrator, I want to deploy APIMan and its components reusing existing infrastructure components where possible (e.g. ElasticSearch), so I can minimize infrastructure components.
+* **UC06** As a cluster administrator, I want API management interface to be able to discover the route to Kibana, so it can be linked to from the management interface. 
+* **UC07** As an API consumer, I want to configure my services to consume other services
+
+## Specification
+
+API services will be scoped to OpenShift projects. Projects will have one-to-one relationship to an APIMan namespace (formally organization). APIMan will utilize namespace
+to manage policy regarding the service.
+ 
+### Service Annotations (and labels?)
+Services intended to be managed by APIMan and exposed as API Endpoints will be annotated<sup>[5](#r5)</sup>
+
+```
+  apiVersion: "v1"
+  kind: "Service"
+  metadata: 
+    annotations: 
+      api.service.kubernetes.io/protocol: REST
+      api.service.kubernetes.io/scheme: http
+      api.service.kubernetes.io/path: cxfcdi
+      api.service.kubernetes.io/description-path: cxfcdi/swagger.json
+      api.service.kubernetes.io/description-language: SwaggerJSON
+```
+
+** do we need labels to generate backlink from apiman management to console service?
+
+### Origin Web Console UI extension
+A custom extension<sup>[6](#r6)</sup> will be created and hooked into the web console to support API endpoints.  The extension
+will allow a user to navigate to the APIManagement interface to:
+* Manage an API endpoint
+* Expose a service as a new API endpoint
+
+The extension will provide the following details to the APIMan gateway:
+* Service name
+* Service namespace
+* User's Oauth token
+
+Calls from the extension to the APIMan gatway will utilize a REST POST call where the oauth token is part of the payload.  It will
```

Comment:
I'm not sure how oauthclient is used in this case.  I was explicitly thinking of the case where a user navigates from the web console to the APIMan management console and trying to avoid a second login which is how we handled auth for Kibana.  The case not addressed in this statement, though I'm not sure it applies, would be a call to the gateway to perform some action for you.